src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -2,96 +2,143 @@
 
 BallPlacementPlayFSM::BallPlacementPlayFSM(TbotsProto::AiConfig ai_config)
     : ai_config(ai_config),
-      pivot_kick_tactic(std::make_shared<WallKickoffTactic>(ai_config)),
-      place_ball_tactic(std::make_shared<PlaceBallTactic>(ai_config)),
-      align_placement_tactic(std::make_shared<PlaceBallMoveTactic>()),
+      align_wall_tactic(std::make_shared<BallPlacementMoveTactic>()),
+      pickoff_wall_tactic(std::make_shared<BallPlacementDribbleTactic>(ai_config)),
+      place_ball_tactic(std::make_shared<BallPlacementDribbleTactic>(ai_config)),
+      align_placement_tactic(std::make_shared<BallPlacementMoveTactic>()),
       retreat_tactic(std::make_shared<MoveTactic>()),
-      move_tactics(std::vector<std::shared_ptr<PlaceBallMoveTactic>>())
+      wait_tactic(std::make_shared<MoveTactic>()),
+      move_tactics(std::vector<std::shared_ptr<BallPlacementMoveTactic>>())
 {
 }
 
-void BallPlacementPlayFSM::kickOffWall(const Update &event)
+void BallPlacementPlayFSM::alignWall(const Update &event)
 {
-    PriorityTacticVector tactics_to_run = {{}};
+    PriorityTacticVector tactics_to_run = {{align_wall_tactic}};
+
+    Point ball_pos           = event.common.world_ptr->ball().position();
+    Rectangle field_boundary = event.common.world_ptr->field().fieldBoundary();
+
+    pickoff_final_orientation = calculateWallPickOffDest(ball_pos, field_boundary).first;
+
+    pickoff_point =
+        ball_pos - Vector::createFromAngle(pickoff_final_orientation).normalize(0.4);
 
-    // setup move tactics for robots away from ball placing robot
-    setupMoveTactics(event);
+    align_wall_tactic->updateControlParams(
+        pickoff_point, pickoff_final_orientation, TbotsProto::DribblerMode::OFF,
+        TbotsProto::BallCollisionType::AVOID, {AutoChipOrKickMode::OFF, 0},
+        TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+        TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE);
+
+    setupMoveTactics(event, event.common.num_tactics -
```

Comment:
needs safety checks that `event.common.num_tactics > tactics_to_run[0].size()`. Otherwise, we risk underflowing into an unsigned type.